src/cleanfigure.m

```diff
@@ -743,89 +752,198 @@ function movePointsCloser(meta, handle)
+                     length(xData)-id_insert(end)];
+
+  % Put the data into one matrix
+  data = [xData(:), yData(:)];
+
+  % Cut the data into segments
+  dataCell   = mat2cell(data, length_segments, 2);
+
+  % Merge the cell arrays
+  dataCell = [dataCell';
+              dataInsert'];
+
+  % Merge the cells back together
+  data     = cat(1, dataCell{:});
+
+  % Set the new (masked) data.
+  set(handle, 'XData', data(:,1));
+  set(handle, 'YData', data(:,2));
+  return;
+end
+% =========================================================================
+function removeData(meta, handle, id_remove)
+  % Removes the data at position id_remove
+
+  % Only do something if id_remove is not empty
+  if isempty(id_remove)
+      return
+  end
   
-      % Rotational matrix with sign flip. It transforms a given vector [a,b] by 
-      % Rotate * [a,b] = [-b,a] as required for calculation of invA  
-      Rotate = [0, -1; 1, 0];   
+  % Extract the data from the current line handle.
+  xData = get(handle, 'XData');
+  yData = get(handle, 'YData');
+  if meta.is3D
+  	zData = get(handle, 'ZData');
+  end
 
+  % Remove the data indicated by id_remove
+  xData(id_remove) = [];
+  yData(id_remove) = [];
+  if meta.is3D
+  	zData(id_remove) = [];
+  end
 
-      % Rather than calculating invA first and then multiply with rhs to obtain 
-      % lambda, directly calculate the respective terms
-      % The upper half of the 2x2 matrix is always the same and is given by:
-      % [-(X4(2)-X3(2)), X4(1)-X3(1)] / detA * rhs
-      % This is a matrix multiplication of the form [1x2] * [2x1] = [1x1]
-      % As we have transposed rhs we can write this as:
-      % rhs * Rotate * (X4-X3) => [nx2] * [2x2] * [2x1] = [nx1]
-      lambda(id_detA, 1) = (rhs(id_detA, :) * Rotate * (X4-X3))./detA(id_detA);
+  % Set the new data.
+  set(handle, 'XData', xData);
+  set(handle, 'YData', yData);
+  if meta.is3D
+  	set(handle, 'zData', zData);
+  end
+  return;    
+end
+% =========================================================================
+function removeNaNs(meta, handle)
+  % Removes superflous NaNs in the data
 
-      % The lower half is dependent on (X2-X1) which is a matrix of size [nx2]
-      % [-(X2(2)-X1(2)), X2(1)-X1(1)] / detA * rhs 
-      % As both (X2-X1) and rhs are matrices of size [nx2] there is no simple 
-      % matrix multiplication leading to a [nx1] vector. Therefore, use the
-      % elementwise multiplication and sum over it
-      % sum( [nx2] * [2x2] .* [nx2], 2) = sum([nx2],2) = [nx1] 
-      lambda(id_detA, 2) = sum(-(X2(id_detA, :)-X1(id_detA, :)) * Rotate .* rhs(id_detA, :), 2)./detA(id_detA);
+  % Extract the data from the current line handle.
+  xData = get(handle, 'XData');
+  yData = get(handle, 'YData');
+  if meta.is3D
+  	zData = get(handle, 'ZData');
+    data  = [xData(:), yData(:), zData(:)];
+  else      
+    data  = [xData(:), yData(:)];
+  end
+
+  % Remove consecutive NaNs
+  id_nan   = any(isnan(data),2);
```

Comment:
Nitpicking mode: I think those `=`s can do with a space before and some of the `,` could stand a space after.